### DISCUSSION TITLE
Function aliases improve issues

### DISCUSSION THREAD

**Discussion by kkk25641463** [databendlabs/databend#15192] [General]
Snowflake [to_geometry()](https://docs.snowflake.com/en/sql-reference/functions/to_geometry) function match with [ST_GEOMETRYFROMWKB](https://docs.snowflake.com/en/sql-reference/functions/st_geometryfromwkb) and [ST_GEOMETRYFROMWKT](https://docs.snowflake.com/en/sql-reference/functions/st_geometryfromwkt).But in databend code, the register_aliases only depend on alias_name which will cause the registered functions to be overwritten. 
```
    pub fn register_aliases(&mut self, fn_name: &str, aliases: &[&str]) {
        for alias in aliases {
            self.aliases.insert(alias.to_string(), fn_name.to_string());
        }
    }
    registry.register_aliases("st_geometryfromwkb", &[
        "to_geometry",
    ]);
    registry.register_aliases("st_geometryfromwkt", &[
        "to_geometry",
    ]);
```
Who have any plans for improvement?

**Comment by sundy-li** [comment]
can we detect the input is wkb or wkt in function execution?

**Reply by kkk25641463** [reply]
Ok，I will implement it separately, not as an alias.
